beacon-chain/core/transition/transition_no_verify_sig.go

```diff
@@ -137,16 +137,6 @@ func CalculateStateRoot(
 	if err != nil {
 		return [32]byte{}, errors.Wrap(err, "could not process block")
 	}
-	if signed.Version() == version.Altair || signed.Version() == version.Bellatrix {
```

Comment:
I added some code that needs to call `ProcessBlockForStateRoot`. I noticed that there were 2 existing sites that called `ProcessBlockForStateRoot`, and they both followed that call w/ the same conditional (essentially checking if the version was not phase0) wrapping the same code (computing and then processing the SyncAggregate). I also checked the specs and noted that, in the specs, `process_sync_aggregate` is invoked by `process_block`[1]. So rather than once againcopy/pasting this chunk of code, I moved it into `ProcessBlockForStateRoot`, and rather than wrap it in a sequence of version checks that would need to be updated for each new fork, I early return from the function when we're on a version past phase0, then after the early return do the > phase0 processing work.
 
[1] https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md#block-processing

Reply:
Thanks for explaining, this makes perfect sense now :-)